policy-controller/k8s/status/src/index.rs

```diff
@@ -91,10 +89,11 @@ pub struct IndexMetrics {
 }
 
 #[derive(Clone, PartialEq)]
-struct HttpRoute {
+struct RouteRef {
     parents: Vec<ParentReference>,
     backends: Vec<BackendReference>,
     statuses: Vec<gateway::RouteParentStatus>,
+    r#type: crate::routes::RouteType,
```

Comment:
I think we can avoid adding this RouteType type altogether. These route refs are stored in a HashMap keyed by NamespaceGroupKindName.  Therefore, we know the kind of the resource from the key.